### PR TITLE
fix #8, fancy escape information printing

### LINKED ISSUE #8: nicer printing of escape information
Something like LoweredCodeUtils's [`print_with_code`](https://juliadebug.github.io/LoweredCodeUtils.jl/stable/edges/).

### PR DESCRIPTION
**Pull request by aviatesk**
Looks like this:
```julia
julia> @analyze_escapes sin(10)
typeof(sin)(_2::Int64 ↑, _3::Float64 ↓)
1 → 1 ─ %1 = Base.sitofp(Float64, _2)::Float64
2 ◌ └──      goto #3 if not false
3 ◌ 2 ─      nothing::Nothing
4 ↑ 3 ┄ %4 = invoke Base.Math.sin(%1::Float64)::Float64
5 ◌ └──      return %4

julia> @analyze_escapes sin(10)
typeof(sin)(_2::Int64 ⤒, _3::Float64 ⤓)
1 ⇥ 1 ─ %1 = Base.sitofp(Float64, _2)::Float64
2 ◌ └──      goto #3 if not false
3 ◌ 2 ─      nothing::Nothing
4 ⤒ 3 ┄ %4 = invoke Base.Math.sin(%1::Float64)::Float64
5 ◌ └──      return %4

julia> @analyze_escapes sin(10)
typeof(sin)(_2::Int64 ⮬, _3::Float64 ▲)
1 ➥ 1 ─ %1 = Base.sitofp(Float64, _2)::Float64
2 ◌ └──      goto #3 if not false
3 ◌ 2 ─      nothing::Nothing
4 ⮬ 3 ┄ %4 = invoke Base.Math.sin(%1::Float64)::Float64
5 ◌ └──      return %4
```

### CODE REVIEW

File: src/EscapeAnalysis.jl [hunk 1]
```diff
@@ -373,12 +373,78 @@ function analyze_escapes(@nospecialize(f), @nospecialize(types=Tuple{});
     interp = EscapeAnalyzer{EscapeState}(interp, nothing, nothing)
 
     code_typed(f, types; optimize=true, world, interp)
-    return interp.source, interp.info
+    return EscapeAnalysisResult(interp.source, interp.info)
 end
 
+# utitlities
+# ==========
```
Comment (aviatesk) [inline src/EscapeAnalysis.jl]:
```suggestion
# utilities
# =========
```

**Comment by aviatesk** [comment]
Hmm, some fonts don't support these characters ? Probably we want more simpler arrows like `↖` for `ReturnEscape`.

**Comment by TH3CHARLie** [comment]
Maybe instead of using one character, directly using abbreviations like `NE, ES, RE, NI`?

**Comment by aviatesk** [comment]
I'd like to stick with arrows.

**Pull request merged**